### ISSUE TITLE
Custom button next to predefined buttons

### ISSUE THREAD

**Issue by karishmasahu** [surveyjs/survey-library#2123]
I want to add custom buttons next to the predefined Next/Complete button. 
So is there a way to achieve it?

**Comment by tsv2013** [comment]
You can use the https://surveyjs.io/Documentation/Library?id=surveymodel#onAfterRenderSurvey event or build your custom navigation as shown in this live sample - https://surveyjs.io/Examples/Library?id=survey-customnavigation&platform=jQuery&theme=modern

**Comment by karishmasahu** [comment]
As per the live demo, We can change the navigation buttons name.
Suppose I want "Send" button and a "Cancel" button at the end of the form... Using the live demo link I will change the "Complete" button name to "send"... But how can I create a "Cancel" button.

**Comment by tsv2013** [comment]
Via adding your markup and binding it to your handler as shown in this sample - https://surveyjs.io/Examples/Library?id=survey-customnavigation&platform=jQuery&theme=modern#content-html

```HTML
        External navigation:
        <span id="surveyProgress"></span>
        <a id="surveyPrev" href="#" onclick="survey.prevPage();">Prev</a>
        <a id="surveyNext" href="#" onclick="survey.nextPage();">Next</a>
        <a id="surveyComplete" href="#" onclick="survey.completeLastPage();">Complete</a>
        <br/>
```

**Comment by shekharkc** [comment]
@tsv2013 

Adding to this thread, quite a few times, a long survey has multiple sections, each section having a set of questions and an intermediate result presented to the user.

Usually, the user is presented with an option to either 
-  'Accept' the result and proceed ahead with the survey 
- Alternately, "reAttempt' that concerned section (**Go Back** to the **Starting** question of _that section_)

This is quite a common **use case.**

With the default options available here ONLY  for adding 'Next' and 'Back' - the survey creator has to add that custom navigation as given in the example presented - involving adding a custom  'Go Back' button in the & adding a clickHandler.  A tedious job for a survey-creator with no programming skills.

It would be a nice feature to add on to the library if you can give an option of adding one more button type - 'Go Back to (QuestionName)' as one of the options, so that the surveycreator can add this button at the intermediate result page, and define which question the user has to be taken back to.

Can this be added to a feature request, if this finds merit in your consideration?

Thanks

**Comment by tsv2013** [comment]
Yes, this scenario makes sense. I think it can include support of the names sections in the survey, These sections should be createable and editable by SurveyJS Creator. Survey navigation should be designed with sections support. Each section can has a summary with results (optional) and optional re-take button.

Unfortunately you are the first in my memory who asks about this feature. We we'll take this in consideration as a feature request, but I can't promise you anything.

At this moment you can create a custom widget that will render results of some previous questions (marked sy some section tag or all the questions between this and previous section) and contain a "re-take" button that will navigate to the first question of the current section. This custom widget will be avialable in the SurveyJS Creator and end-users can add it. Of course it requires coding but this implements exactly your scenario.

**Comment by shekharkc** [comment]
Thank you very much
for your feedback and taking my comments into consideration.
Much appreciated.
